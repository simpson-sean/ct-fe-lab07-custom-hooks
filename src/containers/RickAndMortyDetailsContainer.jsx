import React from 'react';
import { useGetCharacter } from '../hooks/CharacterHooks';
import { Link } from 'react-router-dom';
import Character from '../components/app/characters/Character';


const RickAndMortyDetailsContainer = () => {

    const {loading, character} = useGetCharacter();

    if(loading)return<img src="https://powerusers.microsoft.com/t5/image/serverpage/image-id/118082i204C32E01666789C?v=v2" alt="loading spinner" />

    return (
        <div data-testid="testing-div">
            <Character image={character.image} name={character.name} status={character.status}  />
            <Link to={`/`}>Go Back</Link>
        </div>
    )
    }

export default RickAndMortyDetailsContainer;
