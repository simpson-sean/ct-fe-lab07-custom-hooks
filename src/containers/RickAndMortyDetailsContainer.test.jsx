import React from 'react';
import { render, screen } from '@testing-library/react';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { MemoryRouter } from 'react-router';
import detailData from '../fixtures/detail.json';
import RickAndMortyDetailsContainer from './RickAndMortyDetailsContainer';


const server = setupServer(
  rest.get(`https://rickandmortyapi.com/api/character/:id`, (req, res, ctx) => {
    return res(ctx.json(detailData));
  })
);

describe('RickAndMortyDetails', () => {
  beforeAll(() => server.listen());
  afterAll(() => server.close());

  it('display a single character', async () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/1']}>
        <RickAndMortyDetailsContainer />
      </MemoryRouter>
    );

    screen.getByAltText("loading spinner");

    await screen.findByText('Rick', { exact: false });
    expect(container).toMatchSnapshot();
  });
});