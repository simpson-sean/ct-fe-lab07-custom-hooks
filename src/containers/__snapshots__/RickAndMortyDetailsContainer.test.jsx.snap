// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RickAndMortyDetails display a single character 1`] = `
<div>
  <div
    data-testid="testing-div"
  >
    <img
      alt="Rick Sanchez"
      src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
    />
    <h1>
      Rick Sanchez
    </h1>
    <h3>
      Alive
    </h3>
    <a
      href="/"
    >
      Go Back
    </a>
  </div>
</div>
`;
